### PR TITLE
Implemented "showWhoHasAccess" command

### PR DESCRIPTION
**Pull request by mitar**
Fixes: #1850

### CODE REVIEW

**Comment by garply** [comment]
Hey admins! Can one of you let me know if it's safe to build this pull request? See https://alpha.sandstorm.io/grain/kB7bofATTL2jGuKSFwmqdA/GithubPRCommands for a list of commands.

**Comment by zarvox** [comment]
Garply, ok to test

File: shell/client/grain-client.js [hunk 1]
```diff
@@ -39,6 +39,18 @@ const promptNewTitle = function () {
...
 
+const showWhoHasAccess = function () {
+  const closer = globalTopbar.addItem({
+    name: "who-has-access",
+    template: Template.whoHasAccess,
+    popupTemplate: Template.whoHasAccessPopup,
+    startOpen: true,
+    onDismiss: function () {
+      return "remove";
+    }
```
Comment (zarvox) [inline shell/client/grain-client.js]: The stylechecks want the trailing comma here, so you can add/remove elements to/from the object without needing to change unrelated lines.

**Comment by zarvox** [comment]
Stylecheck comma aside, this looks good to me. :)

Thanks Mitar for implementing this!

File: shell/client/grain-client.js (hunk 1 above)
Comment (mitar) [inline shell/client/grain-client.js]: You do not know that this is invalid JavaScript, no?

Comment (mitar) [inline shell/client/grain-client.js]: Or does transpiling remove it?

**Comment by mitar** [comment]
Updated.

Do you have a changelog file or something somewhere?

**Comment by paulproteus** [comment]
re: changelog: Yes but typically Kenton updates it as part of doing a
release.

If you're interested in writing docs for the new feature, you can find that
in the docs/ directory. Although I'm not yet sure where the docs for this
should go; perhaps we should work on having reference docs (rather than
tutorial-esque docs) that would include this.

**Comment by mitar** [comment]
Docs are already updated as part of this pull request. Only changelog is not. I will leave changelog then untouched.

**Comment by paulproteus** [comment]
Pardon me; I didn't notice the docs changes in here! They look great to me.

File: shell/client/grain-client.js (hunk 1 above)
Comment (zarvox) [inline shell/client/grain-client.js]: Transpiling removes it, and it's the recommended style in the airbnb style guide: https://github.com/airbnb/javascript#commas--dangling

**Comment by mitar** [comment]
What is the process of getting this to be merged?

**Comment by paulproteus** [comment]
Typically:

Wait for Kenton to review it, or attempt to ping him on IRC. See also:
https://github.com/sandstorm-io/sandstorm/wiki/Pull-Request---Code-Review-Procedure

We typically have a multi-day latency for getting things reviewed & merged,
so thanks for your patience!

**Comment by mitar** [comment]
OK, just so that it is not waiting on something from my side. Otherwise no problem.

**Comment by paulproteus** [comment]
For what it's worth, the `startSharing` call has an integration test that covers it. You could add such a test to this branch, although I'm not sure if Kenton would find that a showstopper for a merge.

You can run the tests locally with:

`make test`

But I don't think in principle we're blocked on you here; we're blocked on Kenton at the moment. :)

**Comment by mitar** [comment]
Where does this [click go](https://github.com/sandstorm-io/sandstorm/blob/432be37489b210b19d9677f29befcd5c910a0f76/tests/tests/postmessage-api.js#L104)? I do not find any HTML related to it?

**Comment by paulproteus** [comment]
I suspect​ it is to the grain in a test app:
https://github.com/sandstorm-io/sandstorm/blob/432be37489b210b19d9677f29befcd5c910a0f76/tests/tests/postmessage-api.js#L32

**Comment by mitar** [comment]
Where is the source of that test app?

**Comment by paulproteus** [comment]
Looks like https://github.com/zarvox/sandstorm-test-python based on https://github.com/search?utf8=%E2%9C%93&q=window.parent.postmessage+startSharing&type=Code&ref=searchresults

**Comment by mitar** [comment]
Thanks!

**Comment by paulproteus** [comment]
I guess to succeed at this the way things are now, you would end up submitting a pull request against that Python app to include a call to the new `window.parent` API, or your test suite would end up embedding a link to a different SPK that includes your changes. So I guess you'll probably have to submit a PR against the test app.

**Comment by mitar** [comment]
Yes, that's the plan. I just didn't know where is the code.

**Comment by kentonv** [comment]
Looks good to me but sounds like you're working on a test. Let me know if you want me to merge now or wait.

**Comment by mitar** [comment]
Let me add a test. Will get back.

**Comment by mitar** [comment]
I made: https://github.com/zarvox/sandstorm-test-python/pull/1

Somebody has to compile that and deploy it and tell me the ID.

**Comment by mitar** [comment]
I added a test. But it requires also a change to the test-app ID.

**Comment by zarvox** [comment]
I'd like it if we renamed `showWhoHasAccess` to `showConnectionGraph`, which @mitar you seemed to indicate was okay on IRC.  Let me know if you want to own adding that to this patchset, or if you'd rather I take it on!

I've merged your change in https://github.com/zarvox/sandstorm-test-python/pull/1 and published an updated package for the testsuite to use at https://alpha-hlngxit86q1mrs2iplnx.sandstorm.io/test-7.spk

**Comment by mitar** [comment]
Feel free to do it. You will also have to update the test spkg ID. And can run and test it locally to see if tests even work. I did tests without running them.
